tests/unittests/classification/test_precision_recall.py

```diff
@@ -661,6 +661,37 @@ def test_corner_case():
     assert res == 1.0
 
 
+def test_multiclass_recall_ignore_index():
```

Comment:
looking to the code and the ignore index is already applied in `_multilabel_stat_scores_format` which reduces the preds/target size the same way as the reference metric so calling it with null weights in fact ignores additional index

Reply:
well getting to the root, so the problem is when you try to use the class metric and there is no update called...

Reply:
The problem is we are using sklearn's ```recall_score``` as a reference for our unittests. But sklearn's ```recall_score``` does not support ```ignore_index``` argument. So even if in ```_reference_sklearn_precision_recall_multiclass()``` function we are using ```remove_ignore_index``` function for removing those predictions whose real values are ```ignore_index``` class before passing it to ```recall_score``` function, it does not matter. Because whenever ```average='macro'``` sklearn's ```recall_score``` will always return mean cosidering the total no. of classes. That is the reason why unittests passed in the first place.I think we need to figure out how we want ```MulticlassRecall()``` to behave when ```average = 'macro'``` and ```ignore_index``` is specified like this issue says (in that case we need to modify the unittests) or like it is implemented according to sklearn. I've prepared a [notebook ](https://colab.research.google.com/drive/1YyEZav4b3BY8w48Is8EceSNFoAnYgzXZ?usp=sharing)for explanation. @Borda pls take a look.

Reply:
The problem is we are using sklearn's ```recall_score``` as a reference for our unittests. So even if in ```_reference_sklearn_precision_recall_multiclass()``` function we are using ```remove_ignore_index``` function for removing those predictions whose real values are ```ignore_index``` class before passing it to ```recall_score``` function, it does not matter. Because whenever ```average='macro'``` sklearn's ```recall_score``` will always return mean cosidering the total no. of classes (as we are passing all the classes in ```recall_score()``` function's ```labels``` argument). That is the reason why unittests passed in the first place. I think we need to fix the unittests to take care of ignore_index using sklearn's ```recall_score()``` function's ```labels``` argument. I've prepared a [notebook](https://colab.research.google.com/drive/1YyEZav4b3BY8w48Is8EceSNFoAnYgzXZ?usp=sharing) for explanation. @Borda pls take a look.